pkg/api/types.go

```diff
@@ -451,6 +451,8 @@ type ISCSIVolumeSource struct {
 	IQN string `json:"iqn,omitempty"`
 	// Required: iSCSI target lun number
 	Lun int `json:"lun,omitempty"`
+	// Optional: Defaults to 'default' (tcp). iSCSI interface name that uses an iSCSI transport.
+	ISCSIIface string `json:"iscsiIface,omitempty"`
```

Comment:
@pmorie @anish iscsiadm(8) says `iface = iscsi interface`

```man
-I, --interface=[iface]
              The interface argument specifies the iSCSI interface to use for the operation.  iSCSI interfaces (iface) are defined in /var/lib/iscsi/ifaces.  For  hardware
              iSCSI  (qla4xxx) the iface config must have the hardware address (iface.hwaddress = port's MAC address) and the driver/transport_name (iface.transport_name).
              The iface's name is then the filename of the iface config. For software iSCSI, the iface config must have either the hardware address  (iface.hwaddress),  or
              the network layer's interface name (iface.net_ifacename), and it must have the driver/transport_name

              The  available  drivers/iscsi_transports  are  tcp  (software iSCSI over TCP/IP), iser (software iSCSI over infinniband), or qla4xxx (Qlogic 4XXXX HBAs). The
              hwaddress is the MAC address or for software iSCSI it may be the special value "default" which directs the initiator to not bind the session  to  a  specific
              hardware  resource and instead allow the network or infinniband layer to decide what to do. There is no need to create a iface config with the default behav‐
              ior. If you do not specify a iface, then the default behavior is used.

              As mentioned above there is a special iface name default. There are three others -- cxgb3i, bnx2i and iser, which does not bind the  session  to  a  specific
              card, but will bind the session to the cxgb3i, bnx2i or iser transport. These are experimental and the use is not supported as a stable interface yet.

              In discovery mode multiple interfaces can be specified by passing in multiple -I/--interface instances. For example,

              "iscsiadm -m discoverydb -t st -p ip:port -I iface0 -I iface2 --discover"

              Will direct iscsiadm to setup the node db to create records which will create sessions though the two intefaces passed in.

              In node mode, only a single interface is supported in each call to iscsiadm.

              This option is valid for discovery, node and iface mode.
```